crates/accelerate/src/permutation.rs

```diff
@@ -95,26 +109,30 @@ fn _inverse_pattern(py: Python, pattern: PyArrayLike1<i64>) -> PyResult<PyObject
     Ok(inverse_i64.to_object(py))
 }
 
-/// Sorts the input permutation by iterating through the permutation list
-/// and putting each element to its correct position via a SWAP (if it's not
-/// at the correct position already). If ``n`` is the length of the input
-/// permutation, this requires at most ``n`` SWAPs.
-///
-/// More precisely, if the input permutation is a cycle of length ``m``,
-/// then this creates a quantum circuit with ``m-1`` SWAPs (and of depth ``m-1``);
-/// if the input  permutation consists of several disjoint cycles, then each cycle
-/// is essentially treated independently.
 #[pyfunction]
-#[pyo3(signature = (permutation_in))]
-fn _get_ordered_swap(py: Python, permutation_in: PyArrayLike1<i64>) -> PyResult<PyObject> {
-    let view = permutation_in.as_array();
-    Ok(get_ordered_swap(&view).to_object(py))
+#[pyo3(signature = (pattern))]
+fn _synth_permutation_basic(py: Python, pattern: PyArrayLike1<i64>) -> PyResult<CircuitData> {
+    let view = pattern.as_array();
+    let num_qubits = view.len();
+    Ok(CircuitData::from_standard_gates(
+        py,
+        num_qubits as u32,
+        get_ordered_swap(&view).iter().map(|(i, j)| {
+            (
+                StandardGate::SwapGate,
+                smallvec![],
+                smallvec![Qubit(*i as u32), Qubit(*j as u32)],
+            )
+        }),
+        Param::Float(0.0),
+    )
+    .expect("Something went wrong in Qiskit's Python realm"))
```

Comment:
Based on our offline discussion I think:

```suggestion
    CircuitData::from_standard_gates(
        py,
        num_qubits as u32,
        get_ordered_swap(&view).iter().map(|(i, j)| {
            (
                StandardGate::SwapGate,
                smallvec![],
                smallvec![Qubit(*i as u32), Qubit(*j as u32)],
            )
        }),
        Param::Float(0.0),
    )
```
should work and answer: https://github.com/Qiskit/qiskit/pull/12605/files#r1647033844

Reply:
Of course this will fail rustfmt and I also didn't test it locally, but it's the basic idea.

Reply:
Thanks! Done in b860eb0 🙂